language/tools/move-cli/src/package.rs

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Result};
+use include_dir::{include_dir, Dir, DirEntry};
+use std::{
+    fs::{self, File},
+    io::Write,
+    path::Path,
+};
+use tempfile::tempdir;
+
+use move_lang::{compiled_unit::CompiledUnit, find_filenames, move_compile};
+use stdlib::{COMPILED_EXTENSION, MOVE_EXTENSION};
+use vm::file_format::CompiledModule;
+
+/// Directory name for the package source files under package/<name>
+const PKG_SOURCE_DIR: &str = "source_files";
+/// Directory name for the package binary files under package/<name>
+const PKG_BINARY_DIR: &str = "compiled";
+
+/// Content for the stdlib package
+const PACKAGE_STDLIB: MovePackage = MovePackage::pkg_stdlib();
+/// Content for the nursery package
+const PACKAGE_NURSERY: MovePackage = MovePackage::pkg_nursery();
+
+struct MovePackage {
+    /// Name of the package
+    name: &'static str,
+    /// The directory containing all the .move source files
+    source_dir: Dir<'static>,
+    /// The directory containing all the .mv binary files, if available
+    binary_dir: Option<Dir<'static>>,
+    /// Dependencies
+    deps: &'static [&'static MovePackage],
+}
+
+impl MovePackage {
+    /// Build the stdlib package statically
+    const fn pkg_stdlib() -> Self {
+        MovePackage {
+            name: "stdlib",
+            source_dir: include_dir!("../../stdlib/modules"),
+            binary_dir: Some(include_dir!("../../stdlib/compiled/stdlib")),
+            deps: &[],
+        }
+    }
+
+    /// Build the nursery package statically
+    const fn pkg_nursery() -> Self {
+        MovePackage {
+            name: "nursery",
+            source_dir: include_dir!("../../stdlib/nursery"),
+            binary_dir: None,
+            deps: &[&PACKAGE_STDLIB],
+        }
+    }
+
+    /// Lay down the source files to a given directory
+    pub fn prepare_source_files(&self, dir_path: &Path) -> Result<Vec<String>> {
```

Comment:
I think the code for generating the source files/module files can be a lot simpler if we make it less parameterized

We should just assume our directory structure and always write to it. This goes with removing that command line argument I mentioned earlier